packages/board-server/tests/integration/board-api.test.ts

```diff
@@ -0,0 +1,494 @@
+          );
+        }
+        if (expected.to) {
+          deepStrictEqual(
+            to,
+            expected.to,
+            `Expected to "${JSON.stringify(to)}" to match "${JSON.stringify(expected.to)}" at index ${index}`
+          );
+        }
+        break;
+      }
+      case "graphstart":
+      case "graphend":
+      case "nodestart":
+      case "nodeend": {
+        deepStrictEqual(
+          data.path,
+          expected.path,
+          `Expected path "${JSON.stringify(data.path)}" to match "${JSON.stringify(expected.path)}" at index ${index}`
+        );
+        break;
+      }
+    }
+  }
+};
+
+const scriptedRun = async (
+  boardName: string,
+  script: { inputs?: Record<string, any>; expected: ExpectedResult[] }[],
+  apiKey: string
+) => {
+  let next;
+  for (const [index, { inputs, expected }] of script.entries()) {
+    const inputData = {
+      ...inputs,
+      $key: apiKey,
+      ...(next ? { $next: next } : {})
+    };
+
+    const { statusCode, data: body } = await makeRequest({
+      path: `/boards/@${account.account}/${boardName}.api/run`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: inputData
+    });
+
+    assert.strictEqual(statusCode, 200);
+
+    const events = body.split('\n\n').filter(Boolean).map(event => {
+      const jsonStr = event.replace('data: ', '');
+      return JSON.parse(jsonStr);
+    });
+
+    assertResults(events, expected, index);
+    next = getNext(events[events.length - 1]);
+  }
+};
+
+const getNext = (result?: any) => {
+  if (!result) {
+    throw new Error("No result provided.");
+  }
+  const [type, data, next] = result;
+  if (type === "error") {
+    throw new Error(data.error as string);
+  }
+  if (type === "input") {
+    return next;
+  }
+  if (type === "output" || type === "end") {
+    return undefined;
+  }
+  throw new Error(`Unexpected state type: ${type}`);
+};
+
+suite('Board API Integration tests', async () => {
+// Tests
+  test('GET /boards should list boards', { concurrency: false }, async () => {
+    const { statusCode, data: body } = await makeRequest({
+      path: '/boards',
+      method: 'GET',
+      headers: { 'Content-Type': 'application/json' }
+    });
+    assert.strictEqual(statusCode, 200);
+  });
+
+  test('POST /boards/@test/simple.json should create a new board', { concurrency: false },  async () => {
+    const { statusCode, data: body } = await makeRequest({
+      path: `/boards/@${account.account}/simple.json?API_KEY=${account.api_key}`,
+      method: 'POST',
+      body: readBoard('simple.bgl.json'),
+      headers: { 'Content-Type': 'application/json' }
+    });
+    assert.strictEqual(statusCode, 200);
+    assert.deepStrictEqual(JSON.parse(body), { "created" : `@${account.account}/simple.json` });
+  });
+
+  test('GET /boards/@test/simple.json should get a board', { concurrency: false }, async () => {
```

Comment:
Do these tests need to run in order? Does this test's success depend on the previous on having been run?

Reply:
They do, I'm afraid. Not an ideal state of affairs but it brought the horizon for this test suite a lot closer. I think I can bring a follow up patch to resolve this now that I have a much better understanding of the API.

Reply:
To be clear - setting concurrency false and wrapping them in a suite ensures they run in order, so I'm not leaving it up to chance or interpretation.